lib/vmdb/settings.rb

```diff
@@ -189,6 +188,22 @@ def self.reset_settings_constant(settings)
     end
     private_class_method :reset_settings_constant
 
+    def self.replace_magic_values!(settings, resource)
+      parent_settings = template_settings = nil
+
+      walk(settings) do |key, value, path, owner|
+        case value
+        when RESET_COMMAND
+          parent_settings ||= parent_settings_without_local(resource).load!.to_hash
+          owner[key] = parent_settings.fetch_path(*path)
+        when RESET_ALL_COMMAND
+          template_settings ||= build_template.load!.to_hash
+          owner[key] = template_settings.fetch_path(*path)
```

Comment:
Ignore my implementation of reset_all, because I have no idea what the "real" reset_all is trying to do.

Reply:
@Fryguy ```reset_all>>``` is trying to remove all changes to some key/node and bring ```Settings``` to the initial state (for that key/node) when values coming from ```yml``` file.

```<<reset_all>>``  is quite destructive command - remove all changes without back-up and may be we should just remove it.

For global region: ```setting_changes``` not excluded from replication and, I think, we should ignore both ```<<reset>>``` and ```<<reset_all>>``` on global region

Reply:
@Fryguy I think the comment [here](https://github.com/ManageIQ/manageiq/pull/17563/files#diff-6bbc56fba708243deeb6a7af2b80b06aR26) does a fairly good job of describing what the original intention was.

I guess a better description would be to reset that key to the OOTB value. I think the confusion here is how to make that happen. If I understand correctly there are two options:
1. Explicitly set the value for this resource to the value from the `settings.yml`
    - This option doesn't affect the setting for any other resource under the same parent.
    - What this PR is doing (I think)
2. Remove all customizations for this key
    - This will change the setting for other resources (I don't think this would be do-able with the proposed implementation in this PR)
    - What the original PR did

I think the first option will be the least surprising to the user.

Reply:
Going to continue this conversation below, so as not to split the 2 threads (didn't see the responses here initially) - and also because this is going to get lost on code change